Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Post Item/Components/Post Interaction Bar.swift

```diff
@@ -0,0 +1,157 @@
+    
+    init(post: APIPostView, account: SavedAccount, compact: Bool, voteOnPost: @escaping (ScoringOperation) async -> Void) {
+        self.post = post
+        self.account = account
+        self.compact = compact
+        self.voteOnPost = voteOnPost
+        _dirtyVote = State(initialValue: post.myVote ?? .resetVote)
+        _dirtyScore = State(initialValue: post.counts.score)
+        _dirtySaved = State(initialValue: false)
+        _dirty = State(initialValue: false)
+    }
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            HStack(spacing: 8) {
+                VoteComplex(vote: displayedVote, score: displayedScore, upvote: upvote, downvote: downvote)
+                    .padding(.trailing, 8)
+                SaveButton(saved: false)
+                    .onTapGesture {
+                        Task(priority: .userInitiated) {
+                            await savePost()
+                        }
+                        // ==== TEMPORARY ==== //
+                        isPresentingAlert = true
+                    }
+                    .alert("That feature isn't implemented yet!",
+                           isPresented: $isPresentingAlert) {
+                    }
+                // ==== END TEMPORARY ==== //
+                ReplyButton()
+                    .onTapGesture {
+                        // ==== TEMPORARY ==== //
+                        isPresentingAlert = true
+                    }
+                    .alert("That feature isn't implemented yet!",
+                           isPresented: $isPresentingAlert) {
+                    }
+                // ==== END TEMPORARY ==== //
+                Spacer()
+                infoBlock
+            }
+            .padding(.horizontal)
+            .padding(.vertical, compact ? 2 : 6)
+        }
+        .dynamicTypeSize(compact ? .small : .medium)
+    }
+    
+    // subviews
+    
+    var infoBlock: some View {
+        // post info component
+        HStack(spacing: 8) {
+            HStack(spacing: iconToTextSpacing) {
+                Image(systemName: "clock")
+                Text(getTimeIntervalFromNow(date: post.post.published))
+            }
+            HStack(spacing: iconToTextSpacing) {
+                Image(systemName: "bubble.left")
+                Text(String(post.counts.comments))
+            }
+        }
+        .foregroundColor(.secondary)
+    }
+    
+    // helper functions
+    
+    func upvote() async -> Void {
+        // fake downvote
+        switch (displayedVote) {
+        case .upvote:
+            dirtyVote = .resetVote
+            dirtyScore = displayedScore - 1
+        case .resetVote:
+            dirtyVote = .upvote
+            dirtyScore = displayedScore + 1
+        case .downvote:
+            dirtyVote = .upvote
+            dirtyScore = displayedScore + 2
+        }
+        dirty = true
+        
+        // wait for vote
+        await voteOnPost(.upvote)
```

Comment:
I'm not entirely versed on this, but does this effectively block any further gesture/tap on the upvote button? I'm trying to avoid two different vote-calls going on at the same time

Reply:
@EricBAndrews - I've been looking at the same issue for comments in #156, to avoid triggering repeated calls if the user taps the voting buttons quickly can you add a `guard` at the top of this function to return if we're already _dirty_?

In addition to that inside the `ratePost(...)` function, can you move the `AppConstants.hapticManager.notificationOccurred(.success)` line up so that it happens before we `try await` the response?

With that change the user gets instant feedback for presses which are handled and none for ones we ignore 🤞 

We will look at implementing a queueing system if required in a future PR